compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -978,7 +978,7 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
     }
 
     fn get_mod_child(self, id: DefIndex, sess: &Session) -> ModChild {
-        let ident = self.item_ident(id, sess);
+        let ident = self.opt_item_ident(id, sess).unwrap_or(Ident::empty());
```

Comment:
I think `get_opt_name` should return `Some(kw::Empty)` instead of these hacks.

If not, this needs a comment explaining that synthetic associated items don't have names.

Reply:
Alternatively, I wonder if `ModChild` could be modified to have an `Option<Ident>` instead of an `Ident`. My basic understanding after looking for just a second is that we use that ident for resolution, which we could just filter out if the ident is `None`, since we can never refer to RPITITs by name.